src/LevelZeroIOGroup.cpp

```diff
@@ -0,0 +1,906 @@
+                                  },
+                                  1e6
+                                  }},
+                              {"LEVELZERO::FREQUENCY_MAX_GPU", {
+                                  "Accelerator compute/GPU domain maximum frequency in hertz",
+                                  GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP,
+                                  Agg::average,
+                                  string_format_double,
+                                  {},
+                                  [this](unsigned int domain_idx) -> double
+                                  {
+                                      return this->m_levelzero_device_pool.frequency_max(
+                                                   GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP,
+                                                   domain_idx,
+                                                   geopm::LevelZero::M_DOMAIN_COMPUTE);
+                                  },
+                                  1e6
+                                  }},
+                              {"LEVELZERO::FREQUENCY_MIN_GPU", {
+                                  "Accelerator compute/GPU domain minimum frequency in hertz",
+                                  GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP,
+                                  Agg::average,
+                                  string_format_double,
+                                  {},
+                                  [this](unsigned int domain_idx) -> double
+                                  {
+                                      return this->m_levelzero_device_pool.frequency_min(
+                                                   GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP,
+                                                   domain_idx,
+                                                   geopm::LevelZero::M_DOMAIN_COMPUTE);
+                                  },
+                                  1e6
+                                  }},
+                              {"LEVELZERO::ENERGY", {
+                                  "Accelerator energy in Joules",
+                                  GEOPM_DOMAIN_BOARD_ACCELERATOR,
+                                  Agg::average,
+                                  string_format_double,
+                                  {},
+                                  [this](unsigned int domain_idx) -> double
+                                  {
+                                      return this->m_levelzero_device_pool.energy(
+                                                   GEOPM_DOMAIN_BOARD_ACCELERATOR,
+                                                   domain_idx,
+                                                   geopm::LevelZero::M_DOMAIN_ALL);
+                                  },
+                                  1 / 1e6
+                                  }},
+                              {"LEVELZERO::ENERGY_TIMESTAMP", {
```

Comment:
There are currently 4 timestamp signals.
ENERGY_TIMESTAMP
ACTIVE_TIME_TIMESTAMP
ACTIVE_TIME_COMPUTE_TIMESTAMP
ACTIVE_TIME_COPY_TIMESTAMP

If preferred these can be changed to the form of TIMESTAMP_, but this will requires updates to code in the push_signal function.

Reply:
I think the order here is fine since the timestamp is a supporting qualifier to the primary subject which comes first?